src/Files.App/UserControls/SidebarControl.xaml.cs

```diff
@@ -1099,32 +1006,27 @@ private void SetSize(double val, bool closeImmediatleyOnOversize = false)
 			{
 				var newSize = originalSize + val;
 				if (newSize <= Constants.UI.MaximumSidebarWidth && newSize >= Constants.UI.MinimumSidebarWidth)
-				{
 					OpenPaneLength = newSize; // passing a negative value will cause an exception
-				}
 
-				if (newSize < Constants.UI.MinimumSidebarWidth) // if the new size is below the minimum, check whether to toggle the pane
-				{
-					if (Constants.UI.MinimumSidebarWidth + val <= CompactPaneLength || closeImmediatleyOnOversize) // collapse the sidebar
-					{
-						IsPaneOpen = false;
-					}
-				}
+				if (newSize < Constants.UI.MinimumSidebarWidth &&
+				    (Constants.UI.MinimumSidebarWidth + val <= CompactPaneLength || closeImmediatleyOnOversize)) // if the new size is below the minimum, check whether to toggle the pane
+					IsPaneOpen = false; // collapse the sidebar
 			}
 			else
 			{
-				if (val >= Constants.UI.MinimumSidebarWidth - CompactPaneLength || closeImmediatleyOnOversize)
-				{
-					OpenPaneLength = Constants.UI.MinimumSidebarWidth + (val + CompactPaneLength - Constants.UI.MinimumSidebarWidth); // set open sidebar length to minimum value to keep it smooth
-					IsPaneOpen = true;
-				}
+				if (val >= Constants.UI.MinimumSidebarWidth - CompactPaneLength &&
```

Comment:
this logic needs to be inverted to `val < ...` to match previous behavior

`if (a || b) { ... }` => `if (!a && !b) return`

Reply:
Indeed, thanks for pointing it out.